src/assert_trait.rs

```diff
@@ -103,3 +103,100 @@ macro_rules! assert_trait_super_all {
         $(assert_trait_sub_all!($sub: $super);)+
     };
 }
+
+/// Asserts that the trait is a child of one or more of the other traits.
+///
+/// Related:
+/// - [`assert_impl_any!`]
+///
+/// # Examples
+///
+/// All types that implement [`Copy`] must implement [`Clone`]:
+///
+/// ```
+/// # #[macro_use] extern crate static_assertions; fn main() {}
+/// assert_trait_sub_any!(Copy: Clone);
+/// ```
+///
+/// All types that implement [`Ord`] must implement [`Eq`], but don't have to implement [`Clone`]:
+///
+/// ```
+/// # #[macro_use] extern crate static_assertions; fn main() {}
+/// assert_trait_sub_any!(Ord: Eq, Clone);
+/// ```
+///
+/// The following example fails to compile because neither [`Eq`] nor [`Clone`] are required for
+/// [`PartialOrd`]:
+///
+/// ```compile_fail
+/// # #[macro_use] extern crate static_assertions; fn main() {}
+/// assert_trait_sub_any!(PartialOrd: Eq, Clone);
+/// ```
+///
+/// [`assert_impl_any!`]: macro.assert_impl_any.html
+///
+/// [`Copy`]:       https://doc.rust-lang.org/std/marker/trait.Copy.html
+/// [`Clone`]:      https://doc.rust-lang.org/std/clone/trait.Clone.html
+/// [`Ord`]:        https://doc.rust-lang.org/std/cmp/trait.Ord.html
+/// [`PartialOrd`]: https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html
+/// [`Eq`]:         https://doc.rust-lang.org/std/cmp/trait.Eq.html
+/// [`PartialEq`]:  https://doc.rust-lang.org/std/cmp/trait.PartialEq.html
+#[macro_export]
+macro_rules! assert_trait_sub_any {
+    ($sub:path: $($super:path),+ $(,)?) => {
+        const _: fn() = || {
+            use $crate::_core::marker::PhantomData;
+            use $crate::_core::ops::Deref;
+
+            fn forall<T: $sub>() {
```

Comment:
Small nit: please put an underscore between words.

Reply:
You mean `for_all` ? I'm used to seeing `forall` as a single keyword, e.g. in LaTeX `\forall`, or in Dhall `forall (T: Type) -> ...`, so I don't see it as a mistake, but I have no strong opinion on which one to prefer

Reply:
I'm used to (and prefer) the Rust style of always separating words. I've only seen LaTeX's `\forall` before.